internals/overlord/tlsstate/package_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2025 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package tlsstate_test
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	. "gopkg.in/check.v1"
+)
+
+// Hook up check.v1 into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type tlsSuite struct{}
+
+var _ = Suite(&tlsSuite{})
+
+// GetFakeTime converts a simple string date to the time.Time type for testing purposes.
+func (ts *tlsSuite) GetFakeTime(c *C, date string) time.Time {
```

Comment:
This would be better as a top-level function, as it isn't related to `ts` at all. I think `getFakeTime`.

Reply:
Deleted after refactoring test time. Resolving.